### PR TITLE
Add diagnostic on invalid device list

### PR DESCRIPTION
**Pull request by romanovvlad**

### CODE REVIEW

File: sycl/source/detail/platform_impl.cpp [hunk 1]
```diff
@@ -126,6 +126,9 @@ static std::vector<DevDescT> getAllowListDesc() {
       valuePtr = &decDescs.back().devDriverVer;
       size = &decDescs.back().devDriverVerSize;
       str += sizeof(driverVerStr) - 1;
+    } else {
+      throw sycl::runtime_error("Unrecognized key in device allowlist",
```
Comment (vladimirlaz) [inline sycl/source/detail/platform_impl.cpp:130]: allowlist -> allow list

**Comment by romanovvlad** [comment]
> 
> 
> Should a regression test be supplied?

Ok, will do it, just was hoping that such minor change would be able to sneak without the test :)

File: sycl/source/detail/platform_impl.cpp (hunk 1 above)
Comment (romanovvlad) [reply sycl/source/detail/platform_impl.cpp:130]: It's called allowlist in other messages. So, I think it's better to leave as is.

**Comment by romanovvlad** [comment]
> 
> 
> Should a regression test be supplied?

Ok, will do it, just was hoping that such minor change would be able to sneak without the test :)

> 
> 
> Should a regression test be supplied?

Added a test.

**Pull request merged**